server/src/test/scala/services/crunch/ForecastPlanningToCSVDataTest.scala

```diff
@@ -1,107 +1,272 @@
+          ForecastTimeSlot(d1t0045Millis, 7, 8),
+          ForecastTimeSlot(d1t0100Millis, 9, 10)
+        ),
+        d2t0000Millis -> Seq(
+          ForecastTimeSlot(d1t0000Millis, 1, 2),
+          ForecastTimeSlot(d1t0015Millis, 3, 4),
+          ForecastTimeSlot(d1t0030Millis, 5, 6),
+          ForecastTimeSlot(d1t0045Millis, 7, 8),
+          ForecastTimeSlot(d1t0100Millis, 9, 10)
+        ),
+        d3t0000Millis -> Seq(
+          ForecastTimeSlot(d1t0000Millis, 1, 2),
+          ForecastTimeSlot(d1t0015Millis, 3, 4),
+          ForecastTimeSlot(d1t0030Millis, 5, 6),
+          ForecastTimeSlot(d1t0045Millis, 7, 8),
+          ForecastTimeSlot(d1t0100Millis, 9, 10)
+        )
+      ))
+
+      val result = CSVData.forecastPeriodToCsv(forecast)
+
+      val dt1 = s"""${day1Midnight.getDate()}/${day1Midnight.getMonth()}"""
+      val dt2 = s"""${day2Midnight.getDate()}/${day2Midnight.getMonth()}"""
+      val dt3 = s"""${day3Midnight.getDate()}/${day3Midnight.getMonth()}"""
+      val expected =
+        s"""|,$dt1 - available,$dt1 - required,$dt1 - difference,$dt2 - available,$dt2 - required,$dt2 - difference,$dt3 - available,$dt3 - required,$dt3 - difference
+            |01:00,1,2,-1,1,2,-1,1,2,-1
+            |01:15,3,4,-1,3,4,-1,3,4,-1
+            |01:30,5,6,-1,5,6,-1,5,6,-1
+            |01:45,7,8,-1,7,8,-1,7,8,-1
+            |02:00,9,10,-1,9,10,-1,9,10,-1""".stripMargin
+
+      result === expected
+    }
+  }
+
+  "Given a ForecastPeriod which includes a period that spans a timezone change from BST to UTC " +
+    "Then I should get back a rectangular List of Lists of ForecastTimeSlot Options containing 100 slots each" >> {
+
+    val forecastPeriodDays: Map[MillisSinceEpoch, Seq[ForecastTimeSlot]] = forecastPeriodStartingOnBSTToUTCChange(2).days
+
+    val result: Seq[Int] = CSVData.handleBSTToUTC(forecastPeriodDays).toList.map(_._2.length)
+
+    val expected = 100 :: 100 :: Nil
+
+    result === expected
+  }
+
+  "Given a ForecastPeriod which includes a period that spans a timezone change from BST to UTC " +
+    "Then regular days should have None for the second 1am time period on the timezone change date" >> {
+
+    val forecastPeriod = forecastPeriodStartingOnBSTToUTCChange(2)
+
+    val result = CSVData.rowHeadingsForPeriod(forecastPeriod).take(12)
+
+    val expected = List(
+      "00:00", "00:15", "00:30", "00:45",
+      "01:00", "01:15", "01:30", "01:45",
+      "01:00", "01:15", "01:30", "01:45"
+    )
+    result === expected
+  }
+
+
+  "Given a ForecastPeriod which includes a period that spans a timezone change from BST to UTC " +
+    "Then then row headings should include 1am twice" >> {
+
+    val forecastPeriodDays: Map[MillisSinceEpoch, Seq[ForecastTimeSlot]] = forecastPeriodStartingOnBSTToUTCChange(2).days
+
+    val result: Seq[Int] = CSVData.handleBSTToUTC(forecastPeriodDays).toList.map(_._2.length)
+
+    val expected = 100 :: 100 :: Nil
+
+    result === expected
```

Comment:
This previous test was checking for 1am twice. This test checks there are 100 rows for the forecast period days.